waspc/src/Wasp/Generator/Common.hs

```diff
@@ -12,21 +11,30 @@ import qualified Wasp.SemanticVersion as SV
 -- | Directory where the whole web app project (client, server, ...) is generated.
 data ProjectRootDir
 
--- | Node version that node packages generated by this generator expect.
-nodeVersion :: SV.Version
-nodeVersion = SV.Version 16 0 0 -- Latest LTS version.
-
-nodeVersionBounds :: SV.VersionBounds
-nodeVersionBounds = SV.BackwardsCompatibleWith nodeVersion
-
-npmVersion :: SV.Version
-npmVersion = SV.Version 8 0 0 -- Latest LTS version.
-
-npmVersionBounds :: SV.VersionBounds
-npmVersionBounds = SV.BackwardsCompatibleWith npmVersion
+-- | Range of node versions that node packages generated by this generator work correctly with.
+nodeVersionRange :: SV.Range
+nodeVersionRange =
+  SV.rangeFromVersionsIntersection
+    [ (SV.BackwardsCompatibleWith, latestLTSVersion),
+      (SV.LessThanOrEqual, latestLTSExactVersionThatWeKnowWorks)
+    ]
+  where
+    latestLTSVersion = SV.Version 16 0 0
```

Comment:
maybe rename to `latestLTSNodeVersion`? technically not needed since it is localized, it's just that right below we have the same name but meant for npm, so it might be a bit confusing, this way we make sure the reader knows what it is about